modules/tensor_mechanics/include/materials/IsotropicPlasticityStressUpdate.h

```diff
@@ -29,46 +29,58 @@
  * Press, pg. 146 - 149.
  */
 
-class IsotropicPlasticityStressUpdate : public RadialReturnStressUpdate
+template <bool is_ad>
+class IsotropicPlasticityStressUpdateTempl : public RadialReturnStressUpdateTempl<is_ad>
 {
 public:
   static InputParameters validParams();
 
-  IsotropicPlasticityStressUpdate(const InputParameters & parameters);
+  IsotropicPlasticityStressUpdateTempl(const InputParameters & parameters);
+
+  using Material::_qp;
+  using RadialReturnStressUpdateTempl<is_ad>::_base_name;
+  using RadialReturnStressUpdateTempl<is_ad>::_three_shear_modulus;
 
 protected:
   virtual void initQpStatefulProperties() override;
   virtual void propagateQpStatefulProperties() override;
 
-  virtual void computeStressInitialize(const Real & effective_trial_stress,
-                                       const RankFourTensor & elasticity_tensor) override;
-  virtual Real computeResidual(const Real & effective_trial_stress, const Real & scalar) override;
-  virtual Real computeDerivative(const Real & effective_trial_stress, const Real & scalar) override;
-  virtual void iterationFinalize(Real scalar) override;
-  virtual void computeStressFinalize(const RankTwoTensor & plastic_strain_increment) override;
+  virtual void
+  computeStressInitialize(const GenericReal<is_ad> & effective_trial_stress,
+                          const GenericRankFourTensor<is_ad> & elasticity_tensor) override;
+  virtual GenericReal<is_ad> computeResidual(const GenericReal<is_ad> & effective_trial_stress,
+                                             const GenericReal<is_ad> & scalar) override;
+  virtual GenericReal<is_ad> computeDerivative(const GenericReal<is_ad> & effective_trial_stress,
+                                               const GenericReal<is_ad> & scalar) override;
+  virtual void iterationFinalize(GenericReal<is_ad> scalar) override;
+  virtual void
+  computeStressFinalize(const GenericRankTwoTensor<is_ad> & plastic_strain_increment) override;
 
-  virtual void computeYieldStress(const RankFourTensor & elasticity_tensor);
-  virtual Real computeHardeningValue(Real scalar);
-  virtual Real computeHardeningDerivative(Real scalar);
+  virtual void computeYieldStress(const GenericRankFourTensor<is_ad> & elasticity_tensor);
+  virtual GenericReal<is_ad> computeHardeningValue(GenericReal<is_ad> scalar);
+  virtual GenericReal<is_ad> computeHardeningDerivative(GenericReal<is_ad> scalar);
```

Comment:
pass by const reference!!

Reply:
I'll just have to promise to fix bison because now its failing from a the nonAD class not passing by reference.

Reply:
@dschwen will you approve it this without the bison test passing?  I have an issue to fix bison once this pr goes through:
https://hpcgitlab.hpc.inl.gov/idaholab/bison/-/issues/1426